src/sparsematrix.jl

```diff
@@ -3973,6 +3970,47 @@ function hcat(X::AbstractSparseMatrixCSC...)
     SparseMatrixCSC(m, n, colptr, rowval, nzval)
 end
 
+
+# Efficient repetition of sparse matrices
+
+function Base.repeat(A::AbstractSparseMatrixCSC, m)
+    colptr_source = getcolptr(A)
+    rowval_source = rowvals(A)
+    nzval_source = nonzeros(A)
+
+    nnz_new = nnz(A) * m
+    colptr = similar(colptr_source, length(colptr_source))
+    rowval = similar(rowval_source, nnz_new)
+    nzval = similar(nzval_source, nnz_new)
+
+    colptr[1] = 1
+    for c = 1 : size(A, 2)
+        ptr_res = colptr[c]
+        ptr_source = colptr_source[c]
+        col_length = colptr_source[c + 1] - ptr_source
+        for index_repetition = 0 : (m - 1)
+            row_offset = index_repetition * size(A, 1)
+            stuffcol!(rowval, nzval, ptr_res, rowval_source, nzval_source, ptr_source,
```

Comment:
would it make sense for `stuffcol!` to return the `new ptr_Xi`?

Reply:
I initially decided against this to keep the changes to `stuffcol!` to a minimum in this PR. But I agree, it's a simple enough change and makes the call sites simpler, so since you suggested it I added this in 6db2712

Reply:
thanks!